### PR TITLE
Use & enforce snake_case naming convention on config.json settings

### PR DESCRIPTION
**Pull request by turt2live**
Per internal vote.

**Review with https://github.com/vector-im/element-web/pull/21429**

This change does:
* Rename `ConfigOptions` to `IConfigOptions` to match code convention/style
* Update comments and surrounding documentation
* Define every single option
* Enable a linter to enforce the convention
* Move `IConfigOptions` (and directly related interfaces) to a dedicated file so the linter rules can be applied
* Invent a translation layer for using the new settings with fallbacks (`SnakedObject`)

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://pr8062--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by turt2live** [comment]
https://github.com/matrix-org/matrix-react-sdk/pull/8062/commits/da217d2d68656404aa84b96437323fdc3a437c53 demonstrates the lint rule working:

image

File: docs/settings.md [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
-Settings that support the config level can be set in the config file under the `settingDefaults` key (note that some 
+Settings that support the config level can be set in the config file under the `setting_defaults` key (note that some
 settings, like the "theme" setting, are special cased in the config file):
-```json
+```json5
 {
   ...
-  "settingDefaults": {
+  "setting_defaults": {
     "settingName": true
```
Comment (gsouquet) [inline docs/settings.md:36]:
```suggestion
    "setting_name": true
```

**Review by gsouquet — APPROVED** [review]
Otherwise LGTM

File: docs/settings.md (hunk 1 above)
Comment (turt2live) [reply docs/settings.md:36]: The settings can still be camel case, and most are, unfortunately.

**Pull request merged**